Quadriga/src/main/java/edu/asu/spring/quadriga/dao/impl/publicwebsite/AboutTextDAO.java

```diff
@@ -0,0 +1,56 @@
+package edu.asu.spring.quadriga.dao.impl.publicwebsite;
+
+import java.util.List;
+
+import org.hibernate.HibernateException;
+import org.hibernate.Query;
+import org.springframework.stereotype.Repository;
+
+import edu.asu.spring.quadriga.dao.impl.BaseDAO;
+import edu.asu.spring.quadriga.dao.publicwebsite.IAboutTextDAO;
+import edu.asu.spring.quadriga.dto.AboutTextDTO;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+
+/**
+ * DAO class for public website's about project text.
+ * 
+ * @author Rajat Aggarwal
+ *
+ */
+
+@Repository
+public class AboutTextDAO extends BaseDAO<AboutTextDTO> implements IAboutTextDAO {
+
+    /**
+     * 
+     * @param projectId
+     *            Returns a DTO if already present for the given projectId
+     * @return
+     */
+
+    public AboutTextDTO getDTOByProjectId(String projectId) throws QuadrigaStorageException {
+        AboutTextDTO aboutTextDTO;
+        try {
+            Query query = sessionFactory.getCurrentSession().getNamedQuery("AboutTextDTO.findByProjectId");
+            query.setParameter("projectId", projectId);
+            List<AboutTextDTO> aboutTextDTOList = query.list();
+            if (aboutTextDTOList.size() != 0)
+                aboutTextDTO = aboutTextDTOList.get(0);
+            else
+                return null;
```

Comment:
Both the above were fixed and pushed. Let me know if there are any more
code formatting or jsp formatting that is needed.

Rajat.

Regards,
*Rajat Aggarwal*
Director of Events, Graduate Professional & Student Association *http://*
gpsa.asu.edu
Founder, Graduate Engineering & Science Students Association
*http://www.gessa.org
<http://www.gessa.org/>*
Computer Science Masters Student, Arizona State University





On Tue, Apr 19, 2016 at 1:31 PM, Julia Damerow <notifications@github.com>
wrote:

> In
> Quadriga/src/main/java/edu/asu/spring/quadriga/dao/impl/publicwebsite/AboutTextDAO.java
> <https://github.com/diging/quadriga/pull/199#discussion_r60305381>:
>
> > +     *
> > +     * @param projectId
> > +     *            Returns a DTO if already present for the given projectId
> > +     * @return
> > +     */
> > +
> > +    public AboutTextDTO getDTOByProjectId(String projectId) throws QuadrigaStorageException {
> > +        AboutTextDTO aboutTextDTO;
> > +        try {
> > +            Query query = sessionFactory.getCurrentSession().getNamedQuery("AboutTextDTO.findByProjectId");
> > +            query.setParameter("projectId", projectId);
> > +            List<AboutTextDTO> aboutTextDTOList = query.list();
> > +            if (aboutTextDTOList.size() != 0)
> > +                aboutTextDTO = aboutTextDTOList.get(0);
> > +            else
> > +                return null;
>
> brackets are missing. read the style guide.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/diging/quadriga/pull/199/files/c3204219e03b6c48fbb32bf370f3437a1774f509#r60305381>
>